java-test-samples/schema-and-contract-testing/src/test/java/com/example/ContractTest.java

```diff
@@ -0,0 +1,28 @@
+package com.example;
+
+import java.io.IOException;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+public class ContractTest extends BaseJsonSchemaValidatorTest {
+
+    @Test
+    public void testAddressHasFourFields() throws IOException{
+        JsonNode jsonNode = getJsonNodeFromClasspath("events/customerCreated-event-1.1.0.json");
+        CustomerValidator validator = new CustomerValidator();
+        // Check for business logic in address validation
+        Assertions.assertTrue(validator.isValidAddress(jsonNode));    
```

Comment:
Maybe it's a good idea to also assert on the contents of the address?

Reply:
validation of the content is already on CustomerValidation. Do we need to remove the tool (CustomerValidation) and instead put the assert here?

Reply:
CustomerValidator is not executed as part of the tests. I am not clear as to the purpose of this class. Seems like it's there for completeness.

Reply:
CustomerValidator work as internal validator / business validator class. In most production code, they already have this kind of validator - can be as annotation, or abstract class and just call validate() or isValid() method from there. We want to show that during contract test, we rely on those kind of validation - by having this kind of helper tool or annotation, not just schema. Hope my explanation clear.

Reply:
I have added those CustomerValidator test. Please review again. Thanks!